uint32 dimension
double[] length
double[] radius
